engine/runtime-instrument-common/src/main/java/org/enso/interpreter/service/ExecutionService.java

```diff
@@ -261,6 +262,24 @@ public void execute(
         onExecutedVisualizationCallback);
   }
 
+  /**
+   * Replace the missing arguments of the provided function call with {@link
+   * org.enso.interpreter.node.expression.builtin.Nothing} to make sure that the function call can
+   * be invoked.
+   *
+   * @param functionCall the function call to update
+   */
+  private void substituteMissingArguments(
+      FunctionCallInstrumentationNode.FunctionCall functionCall) {
+    var arguments = functionCall.getArguments();
+    var argumentInfos = functionCall.getFunction().getSchema().getArgumentInfos();
+    for (var i = 0; i < arguments.length; i++) {
+      if (arguments[i] == null && !argumentInfos[i].hasDefaultValue()) {
+        arguments[i] = context.getBuiltins().nothing();
```

Comment:
Here you are modifying content of array in `functionCall`! Next time somebody uses the same instance of `functionCall` your modification will be there. Is that what you want?

Hint: `arguments.clone()` called just before the first modification can be your friend.